content/post/practical-outbreakend.Rmd

```diff
@@ -0,0 +1,408 @@
+$$
+
+Note that the formulated approach is conservative, because every available case is treated as having the potential to generate new secondary cases according to the entire offspring distribution. In practice, however, observed cases towards the end will be secondary cases of some of the earlier cases. Hence, these primary cases will be attributed as having the ability to generate more secondary cases than they actually have in practice. Another important assumption of the method is that all cases are observed: no asymptomatic cases nor under-reporting is taken into account.
+   
+## Required packages
+
+The following packages, available on CRAN, are needed for this practical:
+
+- [`tidyverse`](http://tidyverse.tidyverse.org) A set of packages (aka. the *hadleyverse*) to enhance the necessary data munging
+- [`openxlsx`](https://cran.r-project.org/web/packages/openxlsx/index.html) to read `.xlsx` files without `rJava` dependence
+- [`pbapply`](http://tidyverse.tidyverse.org) Package to get a progress bar for lengthy computations done using `apply`, `sapply`, and  `lapply`.
+
+To install these packages, use `install.packages` and then load them using
+
+```{r, warning=FALSE, message=FALSE}
+suppressPackageStartupMessages(library("tidyverse"))
+suppressPackageStartupMessages(library("openxlsx"))
+suppressPackageStartupMessages(library("pbapply"))
+```
+
+## Data from the MERS-Cov Oubtreak in Korea, 2015
+
+We use the WHO MERS-CoV data available from [http://www.who.int/csr/don/21-july-2015-mers-korea/en/](http://www.who.int/csr/don/21-july-2015-mers-korea/en/) to illustrate the statistical method. For convenience, these data were download and are distributed as part of the RECON learn github account. 
+
+```{r DATAIO, warning=FALSE}
+linelist <- openxlsx::read.xlsx("../../static/data/MERS-CoV-cases-rok-21Jul15.xlsx",startRow=4)
+```
+We skip the first 3 rows as they contain irrelevant header data. After loading the data, additional data munging is needed in order to convert the date-strings to the `Date` class and fill the missing values in the
+`Date.of.symptoms.onset` column as described in the paper.
+
+```{r DATAMUNGING}
+##Convert all columns containing the String "...Date..." to the Date class
+##using the date/month/Year format (in which the data are available)
+linelist <- linelist %>%
+  mutate_if(grepl("Date",names(linelist)), as.Date, format="%d/%m/%Y")
+
+## As written in the @nishiura_etal2016 paper, the missing onset times 
+## are handled as follows: Whenever the date of illness onset was missing, 
+## we substituted it with the date of laboratory confirmation.
+linelist <- linelist %>%
+  mutate(Date.of.symptoms.onset = 
+           if_else(is.na(Date.of.symptoms.onset),
+                   Date.of.laboratory.confirmation, 
+                   Date.of.symptoms.onset))
+```
+
+At the end of the data munging the first three lines in the data look as follows:
+```{r}
+linelist %>% head(n=3)
```

Comment:
The pipe here is uncessary, as you just have one function. `head(linelist, n = 3)` is a bit more readable IMHO :)

Reply:
Probably a matter of taste how much piping one wants to do (if you pipe, you pipe all?). It's sometimes confusing to unwrap parenthesis in some statements and read left to right in others. Still I changed as suggested as a one function call with arguments in order to be clearer.